python/ray/serve/autoscaling_policy.py

```diff
@@ -88,16 +89,36 @@ def _calculate_desired_num_replicas(
     return desired_num_replicas
 
 
+@dataclass
 @PublicAPI(stability="alpha")
-def replica_queue_length_autoscaling_policy(
-    curr_target_num_replicas: int,
-    current_num_ongoing_requests: List[float],
-    current_handle_queued_queries: float,
-    config: Optional[AutoscalingConfig],
-    capacity_adjusted_min_replicas: int,
-    capacity_adjusted_max_replicas: int,
-    policy_state: Dict[str, Any],
-) -> int:
+class AutoscalingContext:
+    """The context for an autoscaling policy call."""
+
+    # The AutoscalingConfig which the deployment started with.
+    config: AutoscalingConfig
+    # The number of replicas that the deployment is currently trying to scale to.
+    current_target_num_replicas: int = 0
+    # List of number of ongoing requests for each replica.
+    current_num_ongoing_requests: List[int] = field(default_factory=list)
+    # The number of handle queued queries, if there are multiple handles, the max
+    # number of queries at a single handle will be passed.
+    current_handle_queued_queries: float = 0.0
+    # The min_replica of the deployment adjusted by the target capacity.
+    capacity_adjusted_min_replicas: Optional[int] = None
+    # The max_replica of the deployment adjusted by the target capacity.
+    capacity_adjusted_max_replicas: Optional[int] = None
+    # State of the policy to be used during the call
+    policy_state: Dict[str, Any] = field(default_factory=dict)
```

Comment:
(more general concern, not about this PR specifically)

Note that this field will be lost if the controller crashes (e.g., the head node goes down).

We should either (1) fix this by checkpointing anytime the policy state changes or (2) clearly document this "gotcha"/limitation

Reply:
Makes sense, I will add a comment here and make sure to note it in the custom autoscaling doc. Can be a follow up feature if users require this to be checkpointed.

Reply:
Why can't we persist state in the GCS? 

We should totally persist the state of the policy (eventually) so that head node restart doesn't reset its state potentially sending it haywire

Reply:
Not that we can't, just feel it's a separable component and doing it as a follow up seems to be easier. Will update this PR to include it.